### PR TITLE
fix: Fix incorrect ratchet behavior on relative source path

### LINKED ISSUE #47: ratchet-from doesn't work on real project
Tried to apply `sorald-buildbreaker` with `ratchet-from` on Spoon, but it's not working as expected, failing to even attempt to repair a newly introduced violation. See https://github.com/INRIA/spoon/pull/3912/checks?check_run_id=2525327603#step:3:78

### PR DESCRIPTION
**Pull request by slarse**
Fix #47 

When the source path (i.e. the one given as input to `runSorald`) is relative, the matching between violation specifiers and changed code doesn't work (as the diff filepaths are absolute).

This PR fixes this by resolving the absolute path to the source in `runSorald`.

Note that there's now quite a lot of repetition in the `main.test.ts` test module. I intend to fix that with a separate refactoring, see #48.

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -56,10 +57,10 @@ export async function runSorald(
     core.info('Attempting repairs');
     for (const [ruleKey, violationSpecs] of keyToSpecs.entries()) {
       core.info(`Repairing violations of rule ${ruleKey}: ${violationSpecs}`);
-      const statsFile = path.join(source.toString(), `${ruleKey}.json`);
+      const statsFile = path.join(sourceAbsPath.toString(), `${ruleKey}.json`);
```
Comment (algomaster99) [inline src/main.ts:60]: `sourceAbsPath` will always be a string if [`path.resolve`](https://nodejs.org/api/path.html#path_path_resolve_paths) doesn't throw an exception so you need not call `toString`.

**Review by algomaster99 — CHANGES_REQUESTED** [review]
Just one small change and then we can decide to merge.

File: src/main.ts (hunk 1 above)
Comment (slarse) [reply src/main.ts:60]: Ah, yes, that is true. I just did a variable replace here, replacing all uses of `source` with `sourceAbsPath`.

**Review by algomaster99 — APPROVED** [review]
Shall we merge now?

**Comment by slarse** [comment]
@algomaster99 Yep :)

**Pull request merged**